src/content/quickstart/kind/_index.md

```diff
@@ -0,0 +1,18 @@
+---
+date: 2020-03-17T13:36:18+01:00
+title: "KIND"
+---
+
+## KIND
+
+[KIND](https://github.com/kubernetes-sigs/kind) is a tool to run local Kubernetes clusters inside Docker container nodes.
+
```

Comment:
Something like this would make it , although, it doesn't work yet... I need to look at what's exactly wrong (may be because we use kindnet)

NOTE: you need docker running on the host

Install kind 0.6.1
```bash
curl -Lo ./kind https://github.com/kubernetes-sigs/kind/releases/download/v0.6.1/kind-$(uname)-amd64
chmod +x ./kind
```

also: Also instructions for kubectl: https://kubernetes.io/docs/tasks/tools/install-kubectl/

```bash
cat > cluster1-config.yaml <<EOF
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
networking:
  podSubnet: 10.244.0.0/16
  serviceSubnet: 100.94.0.0/16
nodes:
- role: control-plane
- role: worker
- role: worker
EOF

./kind create cluster --name=cluster1 --config=cluster1-config.yaml
```

```bash
cat > cluster2-config.yaml <<EOF
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
networking:
  podSubnet: 10.245.0.0/16
  serviceSubnet: 100.95.0.0/16
nodes:
- role: control-plane
- role: worker
- role: worker
EOF

./kind create cluster --name=cluster2 --config=cluster2-config.yaml
```

```bash
cat >cluster3-config.yaml <<EOF
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
networking:
  podSubnet: 10.246.0.0/16
  serviceSubnet: 100.96.0.0/16
nodes:
- role: control-plane
- role: worker
- role: worker
EOF

./kind create cluster --name=cluster3 --config=cluster3-config.yaml

```

```bash
kubectl config get-contexts
```

```
CURRENT   NAME            CLUSTER         AUTHINFO        NAMESPACE
*         kind-cluster1   kind-cluster1   kind-cluster1   
          kind-cluster2   kind-cluster2   kind-cluster2   
          kind-cluster3   kind-cluster3   kind-cluster3   
```

```bash
subctl deploy-broker --kubecontext kind-cluster1
```
```
 ✓ Deploying broker 
 ✓ Creating broker-info.subm file
 ✓ A new IPSEC PSK will be generated for broker-info.subm
```

##ignore this workaround 
# Use this workaround to make sure kind has the right IP address for the broker
#inside the kind docker network: 
#
#```bash
#master_ip=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' cluster1-control-plane | head -n 1)
#echo broker is at: $master_ip
#cp broker-info.subm broker-info.subm.backup
#base64 -d broker-info.subm.backup | sed 's/127.0.0.1/'$master_ip'/g' | base64 -w 0 > broker-info.subm
#```


```bash
subctl join --kubecontext kind-cluster2 broker-info.subm --clusterid cluster2 --disable-nat
```

```
* broker-info.subm says broker is at: https://127.0.0.1:34193
? Which node should be used as the gateway?  [Use arrows to move, type to filter]
> cluster2-worker
  cluster2-worker2
? Which node should be used as the gateway? cluster2-worker
 ✓ Deploying the Submariner operator 
 ✓ Created operator CRDs
 ✓ Created operator namespace: submariner-operator
 ✓ Created operator service account and role
 ✓ Deployed the operator successfully
* Discovering network details
    Discovered network details:
        Network plugin:  generic
        Service CIDRs: [100.95.0.0/16]
        Cluster CIDRs: [10.245.0.0/16]
 ✓ Discovering multi cluster details
* Submariner is up and running
 ✓ Deploying Submariner
```

workaround:
```bash
master_ip=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' cluster1-control-plane | head -n 1)
kubectl get Submariner submariner -n submariner-operator --context kind-cluster2 -o yaml  > subm-cluster2.yaml
sed -i -- "s/brokerK8sApiServer: .*/brokerK8sApiServer: $master_ip:6443/g" subm-cluster2.yaml
kubectl delete Submariner submariner -n submariner-operator --context kind-cluster2
kubectl apply -n submariner-operator --context kind-cluster2 -f subm-cluster2.yaml
```

```bash
subctl join --kubecontext kind-cluster3 broker-info.subm --clusterid cluster3 --disable-nat
```

```
* broker-info.subm says broker is at: https://127.0.0.1:45699
? Which node should be used as the gateway? cluster3-worker
 ✓ Deploying the Submariner operator 
 ✓ Created operator CRDs
 ✓ Created operator namespace: submariner-operator
 ✓ Created operator service account and role
 ✓ Deployed the operator successfully
* Discovering network details
    Discovered network details:
        Network plugin:  generic
        Service CIDRs: [100.96.0.0/16]
        Cluster CIDRs: [10.246.0.0/16]
 ✓ Discovering multi cluster details
* Submariner is up and running
 ✓ Deploying Submariner

```

workaround:
```bash
master_ip=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' cluster1-control-plane | head -n 1)
kubectl get Submariner submariner -n submariner-operator --context kind-cluster3 -o yaml  > subm-cluster3.yaml
sed -i -- "s/brokerK8sApiServer: .*/brokerK8sApiServer: $master_ip:6443/g" subm-cluster3.yaml
kubectl delete Submariner submariner -n submariner-operator --context kind-cluster3
kubectl apply -n submariner-operator --context kind-cluster3 -f subm-cluster3.yaml
```



## verify

```
kubectl config use-context kind-cluster2
kubectl config view --flatten >kubeconfig-cluster2

kubectl config use-context kind-cluster3
kubectl config view --flatten >kubeconfig-cluster3


subctl verify-connectivity kubeconfig-cluster2 kubeconfig-cluster3 --verbose
```